fetching/tarball-fetcher/src/worker/tarballWorker.ts

```diff
@@ -0,0 +1,53 @@
+import { createCafs, getFilePathByModeInCafs } from '@pnpm/store.cafs'
+import { type DependencyManifest } from '@pnpm/types'
+import { parentPort } from 'worker_threads'
+import { Readable } from 'stream'
+import safePromiseDefer from 'safe-promise-defer'
+
+parentPort!.on('message', handleMessage)
+
+interface TarballExtractMessage {
+  type: 'extract'
+  buffer: Buffer
+  cafsDir: string
+}
+
+let cafs: any
+
+async function handleMessage (message: TarballExtractMessage | false): Promise<void> {
+  if (message === false) {
+    parentPort!.off('message', handleMessage)
+    process.exit(0)
+  }
+
+  try {
+    switch (message.type) {
+      case 'extract':
+        const { buffer, cafsDir } = message;
+        if (!cafs) {
+          cafs = createCafs(cafsDir)
+        }
+        const streamForTarball = new Readable({
```

Comment:
You'll want to expressly invoke `gunzipSync(buffer)` rather than using the gunzip stream. The latter offloads to the LibUV thread pool, which will cause thread contention when running inside of a worker. Better to run directly on the current thread.

Reply:
Done